x-pack/test/functional/apps/dashboard/async_search/async_search.ts

```diff
@@ -42,10 +45,47 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
 
     it('timed out should show error', async () => {
       await PageObjects.common.navigateToApp('dashboard');
-      await PageObjects.dashboard.gotoDashboardEditMode('Delayed 15s');
+      await PageObjects.dashboard.loadSavedDashboard('Delayed 15s');
       await PageObjects.header.waitUntilLoadingHasFinished();
       await testSubjects.existOrFail('embeddableErrorLabel');
       await testSubjects.existOrFail('searchTimeoutError');
     });
+
+    it('multiple searches are grouped and only single error popup is shown', async () => {
+      await PageObjects.common.navigateToApp('dashboard');
+      await PageObjects.dashboard.loadSavedDashboard('Multiple delayed');
+      await PageObjects.header.waitUntilLoadingHasFinished();
+      await testSubjects.existOrFail('embeddableErrorLabel');
+      // there should be two failed panels
+      expect((await testSubjects.findAll('embeddableErrorLabel')).length).to.be(2);
+      // but only single error toast because searches are grouped
+      expect((await testSubjects.findAll('searchTimeoutError')).length).to.be(1);
+
+      // check that session ids are the same
+      const getSearchSessionIdByPanel = async (panelTitle: string) => {
+        await dashboardPanelActions.openInspectorByTitle(panelTitle);
+        await inspector.openInspectorRequestsView();
+        const searchSessionId = await (
+          await testSubjects.find('inspectorRequestSearchSessionId')
+        ).getAttribute('data-search-session-id');
+        await inspector.close();
+        return searchSessionId;
+      };
+
+      const panel1SessionId1 = await getSearchSessionIdByPanel('Sum of Bytes by Extension');
```

Comment:
Would it make sense to add tests to `test/plugin_functional/test_suites/data_plugin/session.ts` to make sure that sessions are being created once per load \ reload?

It was what uncovered the problematic behaviors in Dashboard in the first place.

Reply:
I will take a look 👍 
but would like to do it in separate pr, 
really want to unblock a bunch of follow up prs with merging this :(